src/plugins/interactive_setup/server/verification_code.test.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { loggingSystemMock } from 'src/core/server/mocks';
+
+import { VERIFICATION_CODE_LENGTH } from '../common';
+import { VerificationCode } from './verification_code';
+
+const loggerMock = loggingSystemMock.createLogger();
+
+describe('VerificationCode', () => {
+  it('should generate a 6 digit code', () => {
+    for (let i = 0; i < 10; i++) {
+      const { code } = new VerificationCode(loggerMock);
+      expect(code).toHaveLength(VERIFICATION_CODE_LENGTH);
+      expect(code).toEqual(expect.stringMatching(/^[0-9]+$/));
+    }
+  });
+
+  it('should verify code correctly', () => {
+    const verificationCode = new VerificationCode(loggerMock);
+
+    expect(verificationCode.verify(undefined)).toBe(false);
+    expect(verificationCode.verify('')).toBe(false);
+    expect(verificationCode.verify('invalid')).toBe(false);
+    expect(verificationCode.verify(verificationCode.code)).toBe(true);
+  });
+
+  it('should track number of failed attempts', () => {
+    const verificationCode = new VerificationCode(loggerMock);
+
+    verificationCode.verify('invalid');
+    verificationCode.verify('invalid');
+    verificationCode.verify('invalid');
+    expect(verificationCode['failedAttempts']).toBe(3); // eslint-disable-line dot-notation
```

Comment:
__question:__ why not rely on public `remainingAttempts` instead (here and in tests below)?

Reply:
Because then I need to calculate the difference myself. Here it maps directly to the number of times I called verify which is easier to see in the test

Reply:
Yeah, but it's a `private` field and wouldn't even work at all if you switch to the native JS private fields.